modules/affile/affile-grammar.ym

```diff
@@ -257,12 +256,34 @@ multi_line_option
 	  {
             GError *error = NULL;
 
-	    CHECK_ERROR_GERROR(log_proto_multi_line_server_options_set_garbage(&last_log_proto_options->super, $3, &error), @3, error, "error compiling multi-line regexp");
-	    free($3);
+	    	CHECK_ERROR_GERROR(log_proto_multi_line_server_options_set_garbage(&last_log_proto_options->super, $3, &error), @3, error, "error compiling multi-line regexp");
+	    	free($3);
 	  }
 	| KW_MULTI_LINE_TIMEOUT '(' nonnegative_integer ')'
 	  {
-	    file_reader_options_set_multi_line_timeout(last_file_reader_options, ($3 * 1000));
+	    	file_reader_options_set_multi_line_timeout(last_file_reader_options, ($3 * 1000));
+	  }
+	;
+
+multi_line_option_change
```

Comment:
This solution can work, but you are duplicating the non-timeout related options, which is harder to maintain.

Instead, please partition the `multi_line_option`. `multi_line_option = multi_line_option_without_timeout + timeout`. Then you can `use multi_line_option_timeout` in the logpipe source.

Reply:
fixes: https://github.com/syslog-ng/syslog-ng/issues/3178
Made the necessary changes to the code

Reply:
Thanks!